_sips/minutes/2019-06-08-sip-minutes.md

```diff
@@ -0,0 +1,287 @@
+
+Presented by Oliver
+
+Miles: the example is too basic, no recursion: can we do typeclass derivation with this? (need recursion)
+
+Adriaan: how staging works must be clear to the SIP committee and to users, even if it targets advanced users
+Adriaan has concerns about sneaking whitebox macro style problems back into the compiler
+Adriaan/Seb: inline is simpler
+  * Olivier: I disagree, you can't understand a large inline method, because it's entangled with the optimizer
+  * Martin: with staging you can println half-way into staging
+  * Martin: the ideal is that inline desugars to staging
+  * Nicolas: except staging requires separate compilation (because we have no interpreter), inline doesn't
+  * (Miles: or we can make an interpreter for a subset of the language)
+  * Nicolas also has some ingenious half-way between inline and staging, that he mentioned (too) quickly
+
+### More general meta-programming talks
+
+Nicolas/Oliver present 3 "tiers" of meta-programming: inline, staging, dotty reflect
+
+At the end it becomes clear that these aren't desugarings from one to another (e.g. staging doesn't desugar
+to dotty reflect, it's its own thing, idem for inline to staging).
+
+But even at the most powerful, dotty reflect, you cannot typecheck (or untypecheck), you only have typed trees.
+
+No precise hanging questions, follow-up work or decisions, except:
+
+Miles: how long to finish everything?
+  Nicholas: hopefully by the end of the summer
+
+### Typeclass Derivation
+
+[https://dotty.epfl.ch/docs/reference/contextual/derivation.html](https://dotty.epfl.ch/docs/reference/contextual/derivation.html)
+
+Presented by Miles.
+
+For typeclasses that are monoidal (e.g. Monoid, Eq/Eqv, Show) just pass the summoned TC instances to the erased
+infrastructure.
+
+There is a "derived" magic method name on TC companions that matches the "derives" keyword that you can put at
+the end of your case class.
+
+Seth: no handling of higher-kinded types?
+  * Miles: no need at the compiler/scala.deriving level, just a small kernel, and a veneer in Shapeless 3
+  * Miles: AnyKind is not enough to do the necessary kind-polymorphism
+
+Martin: But we said we weren't adding anything to the standard library in 3.0
+  * (Someone): We can still add it to the 2.14 and 3.0 standard library
+
+### Erased Terms
+
+[https://dotty.epfl.ch/docs/reference/metaprogramming/erased-terms.html](https://dotty.epfl.ch/docs/reference/metaprogramming/erased-terms.html)
+
+Olivier: Are they necessary?
+  * Yes's and No's
+  * Seb: in Scala.js I could use the
+  * Seth: should we ask the community?
+  * Miles: no, because you'll always find someone that says yes
+Decision: Find use cases and test performance
+Seth: Defer to 3.1+?
+  (Someone): Yes, it doesn't need to be in 3.0
+(Someone): Maybe it's superseded by @compileTimeOnly, that's already in 2.13?
+(A few): Should we delete the code or put it under a flag?
+  Guillaume: If we don't delete it it'll be in TASTY forever
```

Comment:
This sublist does not render well. All the lines are mashed together as a single line. (at least in GitHub's preview)

Reply:
Indeed, thanks for the review!